File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/MaxlengthValidationTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class MaxlengthValidationTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());

    @Test
    public void testShorterIsValidPasses() {
        // shorter is valid
        final var schema = MaxlengthValidation.MaxlengthValidation1.getInstance();
        schema.validate(
            "f",
            configuration
        );
    }

    @Test
    public void testExactLengthIsValidPasses() {
        // exact length is valid
        final var schema = MaxlengthValidation.MaxlengthValidation1.getInstance();
        schema.validate(
            "fo",
            configuration
        );
    }

    @Test
    public void testTooLongIsInvalidFails() {
        // too long is invalid
        final var schema = MaxlengthValidation.MaxlengthValidation1.getInstance();
        try {
            schema.validate(
                "foo",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException | InvalidTypeException ignored) {
            ;
        }
    }

    @Test
    public void testIgnoresNonStringsPasses() {
        // ignores non-strings
        final var schema = MaxlengthValidation.MaxlengthValidation1.getInstance();
        schema.validate(
            100,
            configuration
        );
    }

    @Test
    public void testTwoSupplementaryUnicodeCodePointsIsLongEnoughPasses() {
        // two supplementary Unicode code points is long enough
        final var schema = MaxlengthValidation.MaxlengthValidation1.getInstance();
        schema.validate(
            "💩💩",
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/MaxlengthValidationTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class MaxlengthValidationTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testShorterIsValidPasses() {
        // shorter is valid
        val schema = MaxlengthValidation.MaxlengthValidation1.getInstance()
        schema.validate(
            "f",
            configuration
        )
    }

    @Test
    fun testExactLengthIsValidPasses() {
        // exact length is valid
        val schema = MaxlengthValidation.MaxlengthValidation1.getInstance()
        schema.validate(
            "fo",
            configuration
        )
    }

    @Test
    fun testTooLongIsInvalidFails() {
        // too long is invalid
        val schema = MaxlengthValidation.MaxlengthValidation1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "foo",
                    configuration
                )
            }
        )
    }

    @Test
    fun testIgnoresNonStringsPasses() {
        // ignores non-strings
        val schema = MaxlengthValidation.MaxlengthValidation1.getInstance()
        schema.validate(
            100,
            configuration
        )
    }

    @Test
    fun testTwoSupplementaryUnicodeCodePointsIsLongEnoughPasses() {
        // two supplementary Unicode code points is long enough
        val schema = MaxlengthValidation.MaxlengthValidation1.getInstance()
        schema.validate(
            "💩💩",
            configuration
        )
    }
}
